cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+}
+
+// Type declaration and functions for sorting the pod PS output
+type podPsSorted []podPsJSONParams
+
+func (a podPsSorted) Len() int      { return len(a) }
+func (a podPsSorted) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+type podPsSortedId struct{ podPsSorted }
+
+func (a podPsSortedId) Less(i, j int) bool { return a.podPsSorted[i].ID < a.podPsSorted[j].ID }
+
+type podPsSortedNumber struct{ podPsSorted }
+
+func (a podPsSortedNumber) Less(i, j int) bool {
+	return len(a.podPsSorted[i].CtrsInfo) < len(a.podPsSorted[j].CtrsInfo)
+}
+
+type podPsSortedName struct{ podPsSorted }
+
+func (a podPsSortedName) Less(i, j int) bool { return a.podPsSorted[i].Name < a.podPsSorted[j].Name }
+
+type podPsSortedStatus struct{ podPsSorted }
+
+func (a podPsSortedStatus) Less(i, j int) bool {
+	return a.podPsSorted[i].Status < a.podPsSorted[j].Status
+}
+
+var (
+	podPsFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "cgroup",
+			Usage: "Print the Cgroup information of the pod",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-names",
+			Usage: "Display the container names",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-ids",
+			Usage: "Display the container UUIDs. If no-trunc is not set they will be truncated",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-status",
+			Usage: "Display the container status",
+		},
+		cli.StringFlag{
+			Name:  "filter, f",
+			Usage: "Filter output based on conditions given",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "Pretty-print pods to JSON or using a Go template",
+		},
+		cli.BoolFlag{
+			Name:  "labels",
+			Usage: "Print the labels of the pod",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "Display the extended information",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Print the numeric IDs of the pods only",
+		},
+		cli.StringFlag{
+			Name:  "sort",
+			Usage: "Sort output by id, name, or number",
+			Value: "name",
+		},
+	}
+	podPsDescription = "Prints out information about pods"
+	podPsCommand     = cli.Command{
+		Name:                   "ps",
+		Aliases:                []string{"ls", "list"},
+		Usage:                  "List pods",
+		Description:            podPsDescription,
+		Flags:                  podPsFlags,
+		Action:                 podPsCmd,
+		UseShortOptionHandling: true,
+	}
+)
+
+func podPsCmd(c *cli.Context) error {
+	if err := validateFlags(c, podPsFlags); err != nil {
+		return err
+	}
+
+	if err := podCheckFlagsPassed(c); err != nil {
+		return errors.Wrapf(err, "error with flags passed")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+
+	defer runtime.Shutdown(false)
+
+	if len(c.Args()) > 0 {
+		return errors.Errorf("too many arguments, ps takes no arguments")
+	}
+
+	format := genPodPsFormat(c.String("format"), c.Bool("quiet"), c.Bool("cgroup"), c.Bool("labels"), c.Bool("ctr-names"), c.Bool("ctr-ids"), c.Bool("ctr-status"))
```

Comment:
Should this take a podPsOptions instead, so we don't have to modify both when we add new parameters?

Reply:
To do so I'd pass in cli.Context instead, so as to not have excess info in podPsOptions that gets translated directly to json. I took this route, because it's passing by pointer so it is not that much overhead, besides the querying of c.*("*")

Reply:
I'm mostly worried about the maintenance burden of having to modify the long arguments list and podPsOptions struct both when adding new options